src/main/steps/how-do-you-want-to-apply/post.ts

```diff
@@ -0,0 +1,33 @@
+import autobind from 'autobind-decorator';
+import { NextFunction, Response } from 'express';
+
+import { ApplicationType, YesOrNo } from '../../app/case/definition';
+import { AppRequest } from '../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../app/controller/PostController';
+import { Form } from '../../app/form/Form';
+
+@autobind
+export default class HowDoYouWantToApplyPostController extends PostController<AnyObject> {
+  constructor(protected readonly form: Form) {
+    super(form);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response, next?: NextFunction): Promise<void> {
+    const { saveAndSignOut, saveBeforeSessionTimeout, _csrf, ...formData } = this.form.getParsedBody(req.body);
+
+    if (
+      formData.applicationType === ApplicationType.SOLE_APPLICATION &&
+      req.session.userCase.applicant2AddressPrivate === YesOrNo.YES
+    ) {
+      req.body.applicant2Address1 = null;
```

Comment:
not sure why but i feel like this is not needed (due to getUnreachableAnswers) or could be moved to the to-api-format check.

Reply:
not a big deal but im getting that feeling.

Reply:
This was A/C 5 on NFDIV-778 - happy to move this if it isn't required